### PR TITLE
Improve verification code fetching logging

### PR DESCRIPTION
**Pull request by sergiishamrai-okta**

### CODE REVIEW

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 1]
```diff
@@ -79,27 +82,35 @@ public void waitForWebElementDisplayed(WebElement webElement) {
...
-            try { Thread.sleep(500); } catch (InterruptedException e) { e.printStackTrace(); }
+        int totalRetryCount = getRetryCountDuringVerificationCodeFetching();
+        int tryCounter = 0;
+        while (tryCounter < totalRetryCount && code == null) {
+            waitForNextTry();
             String sms = Page.getA18NClient().getLatestSmsContent(Page.getA18NProfile());
             code = StringUtils.substringBetween(sms, "code is ", ".");
-            retryCount--;
+            if(code == null) {
+                logger.info("Attempt {} of {} SMS fetching failed.", tryCounter, totalRetryCount);
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:92]: This should be a warn level log.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 2]
```diff
@@ -79,27 +82,35 @@ public void waitForWebElementDisplayed(WebElement webElement) {
...
-            try { Thread.sleep(500); } catch (InterruptedException e) { e.printStackTrace(); }
+        int totalRetryCount = getRetryCountDuringVerificationCodeFetching();
+        int tryCounter = 0;
+        while(tryCounter < totalRetryCount && email == null) {
+            waitForNextTry();
             email = Page.getA18NClient().getLatestEmailContent(Page.getA18NProfile());
-            if(email != null) {
-                break;
+            if(email == null) {
+                logger.info("Attempt {} of {} email fetching failed.", tryCounter, totalRetryCount);
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:109]: We should change it to warn.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 3]
```diff
@@ -110,13 +121,43 @@ public String fetchCodeFromRegistrationEmail(String emailContent) {
         return matcher.find() ? matcher.group(1) : null;
     }
 
+    private void waitForNextTry() {
+        try {
+            Thread.sleep(getSleepDurationDuringVerificationCodeFetching());
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:128]: We should add an ERROR logging here.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 4]
```diff
@@ -110,13 +121,43 @@ public String fetchCodeFromRegistrationEmail(String emailContent) {
...
-    public void doNothing() {
-        try { Thread.sleep(500); } catch (InterruptedException e) { e.printStackTrace(); }
+    int getRetryCountDuringVerificationCodeFetching() {
+        int retry = 10;
+        try {
+            retry = Integer.parseInt(
+                    System.getenv().getOrDefault("OKTA_E2E_VERIFICATION_CODE_RETRY_COUNT", retry + "")
+            );
+        } catch (NumberFormatException e) {
+            logger.debug("Fail to parse OKTA_E2E_VERIFICATION_CODE_RETRY_COUNT. Defaulting to {}.", retry, e);
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:145]: Please change it to ERROR level.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 5]
```diff
@@ -110,13 +121,43 @@ public String fetchCodeFromRegistrationEmail(String emailContent) {
...
+    }
+
+    int getSleepDurationDuringVerificationCodeFetching() {
+        int value = 100;
+        try {
+            value = Integer.parseInt(
+                    System.getenv().getOrDefault("OKTA_E2E_VERIFICATION_CODE_SLEEP_DURATION", value + "")
+            );
+        } catch (NumberFormatException e) {
+            logger.debug("Fail to parse OKTA_E2E_VERIFICATION_CODE_SLEEP_DURATION. Defaulting to {}.", value, e);
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:158]: We should change this to ERROR/WARN level.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 6]
```diff
@@ -110,13 +121,43 @@ public String fetchCodeFromRegistrationEmail(String emailContent) {
...
     public String fetchCodeFromPasswordResetEmail(String emailContent) {
         Pattern pattern = Pattern.compile("Enter a code instead: (\\d{6})");
         Matcher matcher = pattern.matcher(emailContent);
         return matcher.find() ? matcher.group(1) : null;
     }
 
-    public void doNothing() {
-        try { Thread.sleep(500); } catch (InterruptedException e) { e.printStackTrace(); }
+    int getRetryCountDuringVerificationCodeFetching() {
+        int retry = 10;
```
Comment (arvindkrishnakumar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:139]: 10 would probably introduce too much failures (flakiness) in ITs, I fear. Thoughts?

**Review by arvindkrishnakumar-okta — CHANGES_REQUESTED** [review]
Left few minor comments.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java (hunk 6 above)
Comment (sergiishamrai-okta) [reply samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:139]: Usually, it takes 2-3 seconds to get a code from sms/email. Using default value for OKTA_E2E_VERIFICATION_CODE_RETRY_COUNT=10 and OKTA_E2E_VERIFICATION_CODE_SLEEP_DURATION=2000 we have 20secs to get a verification code. Maybe it's better to increase OKTA_E2E_VERIFICATION_CODE_SLEEP_DURATION up to 3-4sec?

**Review by laura-rodriguez — APPROVED** [review]

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Pull request merged**